src/Provider/Miscellaneous.php

```diff
@@ -242,14 +242,30 @@ public static function boolean($chanceOfGettingTrue = 50)
         return self::numberBetween(1, 100) <= $chanceOfGettingTrue;
     }
 
+    /**
+     * Return a string representing a big number, which size is $n * 4 bytes ($n * 32 bits)
+     *
+     * @param int $n multiplier for numberBetween.
+     *
+     * @return string
+     *
+     * @example "38921908753913698485999935526730716"
+     */
+    public static function multiplyNumberBetweenNTimes($n)
+    {
+        return array_reduce(range(1, $n), static function ($carry, $item) {
+            return bcmul($carry, self::numberBetween());
+        }, '1');
```

Comment:
This function is heavily biased, which is easy to see by performing an exhaustive search for a smaller number range:

```php
<?php
for ($a = 0; $a < 16; $a++) 
for ($b = 0; $b < 16; $b++) 
for ($c = 0; $c < 16; $c++) 
@$results[$a * $b * $c]++;

var_dump($results);
```

If any of the multiplied numbers is zero, the result will also be zero, which makes zero the most likely result. But the others are not uniformly distributed either:

https://3v4l.org/VJnZK